### PR TITLE
[platform] NXP platform toolchain cmake file update

### PR DESCRIPTION
**Pull request by TE-N-ElvenWang**
Add link target finding folder and correct readline target variable.
Added the arm toolchain file. This enabled the cross compile based on NXP
i.MX6ULL Linux SDK.
TEST:  source ${PATH_TO_NXP_LINUX_SDK}/environment-setup-cortexa53-crypto-poky-linux
            ./script/cmake-build -DOTBR_BORDER_ROUTING=ON -DOTBR_REST=ON
            -DOTBR_WEB=ON -DOTBR_BACKBONE_ROUTER=ON
            -DOT_BACKBONE_ROUTER_MULTICAST_ROUTING=ON -DBUILD_TESTING=OFF
            -DOTBR_DBUS=ON -DOTBR_DNSSD_DISCOVERY_PROXY=ON
            -DOTBR_SRP_ADVERTISING_PROXY=ON -DOT_THREAD_VERSION=1.2
            -DOTBR_INFRA_IF_NAME=mlan0
            -DCMAKE_TOOLCHAIN_FILE=./examples/platforms/nxp/linux-imx/arm.cmake

### CODE REVIEW

**Comment by TE-N-ElvenWang** [comment]
@jwhui could you please help to review this change?
Thanks!

**Comment by TE-N-ElvenWang** [comment]
Hi @jwhui  and @bukepo , could you please help to merge this? Since we encourage customer to build otbr for i.MX8 and i.MX6 directly based on community codes. This patch fixed the arm64 new SDK toolchain configuration bugs and add arm32 SDK toolchain configuration to support imx6 series SoC.
Thanks.

**Review by bukepo — APPROVED** [review]

File: examples/platforms/nxp/linux-imx/arm.cmake [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(CMAKE_SYSTEM_NAME Linux)
+set( CMAKE_SYSTEM_PROCESSOR arm )
```
Comment (jwhui) [inline examples/platforms/nxp/linux-imx/arm.cmake:29]: Remove the leading and trailing spaces here and all other lines below?

Comment (TE-N-ElvenWang) [reply examples/platforms/nxp/linux-imx/arm.cmake]: Thanks, @jwhui . I've updated the PR based on the comments. Please review the updated one.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**